crates/upload/src/database.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::{anyhow, bail, Result};
+use reqwest::blocking::Client;
+use serde::{de::DeserializeOwned, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+
+/// A simple HTTP wrapper for communicating with an ElasticSearch database.
+pub struct Database {
+    url: String,
+    dryrun: bool,
```

Comment:
Can we name this `dry_run` / `--dry-run` here and everywhere else?

Precedent, at least from `cargo publish`:

```
       --dry-run
           Perform all checks without uploading.
```